app/src/main/kotlin/za/foundation/praekelt/mama/app/activity/MainActivity.kt

```diff
@@ -60,51 +69,31 @@ public class MainActivity : AppCompatActivity(), AnkoLogger {
         navigationView = this.nav_view
         viewPager = this.viewpager
         tabLayout = this.tabs
-        getActivityComponent().inject(this)
     }
 
     override fun onResume() {
         super<AppCompatActivity>.onResume()
+        activityComp().inject(this)
         println("resuming")
-        val networkObs: Observable<Boolean> = Observable
-                .just(connectivityManager.getActiveNetworkInfo())
-                .map { networkInfo -> networkInfo?.isConnected() ?: false }
 
         networkObs.filter { !it }
                 .subscribe { noInternetSnackBar() }
 
-        val hasRepoObs: Observable<Boolean> = networkObs.filter { it }
-                .map { SharedPrefsUtil.getCommitFromSharedPrefs(this) }
-                .map { it != "" }
-
-        hasRepoObs.filter { !it }
-                .doOnNext { info("repo doesn't exist") }
-                .doOnNext { println("repo doesn't exist") }
-                .flatMap { ucdService.cloneRepo() }
-                .doOnNext { DBTransaction.saveRepo(it) }
-                .doOnNext { SharedPrefsUtil.saveCommitToSharedPrefs(this, it.commit) }
-                .doOnNext { println("SP saved") }
-//                .flatMap { (viewPager.getAdapter() as CategoryPageAdapter).refresh() }
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe { getActivityComponent().inject(this) }
-
-        hasRepoObs.filter { it }
-                .doOnNext { info("repo exists, checking for update") }
-                .flatMap { ucdService.getRepoStatus() }
-                .filter { SharedPrefsUtil.getCommitFromSharedPrefs(this) != it.commit }
-                .doOnNext { info("getting update") }
-                .flatMap {
-                    ucdService.pullRepo(SharedPrefsUtil.getCommitFromSharedPrefs(this))
-                }
-                .doOnNext { DBTransaction.saveRepoPull(it) }
-                .doOnNext { println("saved update data") }
-                .doOnNext { SharedPrefsUtil.saveCommitToSharedPrefs(this, it.commit) }
-                .flatMap { (viewPager.getAdapter() as CategoryPageAdapter).refresh() }
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe{ getActivityComponent().inject(this) }
+        val sub = Observers.create<Any>(
+                { evt -> activityComp().inject(this) },
+                { err -> info("Error connecting to network") })
+
+        subscriptions.add(cloneObs.subscribe(sub))
+        subscriptions.add(updateObs.subscribe(sub))
         println("end resuming")
     }
 
+    override fun onPause() {
```

Comment:
Would it be possible to subscribe/unsubscribe in onStart/Stop rather. onResume/Pause gets called a lot...